server/src/main/java/org/opensearch/gateway/ReplicaShardBatchAllocator.java

```diff
@@ -168,6 +168,31 @@ public void allocateUnassignedBatch(List<ShardRouting> shardRoutings, RoutingAll
         logger.trace("Finished shard allocation execution for unassigned replica shards: {}", shardRoutings.size());
     }
 
+    protected void allocateUnassignedBatchOnTimeout(List<ShardRouting> shardRoutings, RoutingAllocation allocation) {
+        Set<ShardId> shardIdsFromBatch = new HashSet<>();
+        for (ShardRouting shardRouting : shardRoutings) {
```

Comment:
The batch would be updated in that case ([code link](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/gateway/ShardsBatchGatewayAllocator.java#L357)), right? Or am I missing something?

Reply:
So if the decision is throttle, it will be removed from unassigned iterator but it would still be part of batch. This can happen today as well. It is mimicking throttling behavior. The only concern I have is sometimes after multiple attempts of throttling, it is not retried automatically and requires manual reroute. Could we end into that issue here? @imRishN

Reply:
Throttling is not considered as a failed allocation, right? Currently, a shard's allocation is marked as failed only during apply failed shard event. If a shard fails 5 times, it requires manual reroute with retry failed marker (this behaviour remains unchanged, aligns with existing process?)